### PR TITLE
Set Pdeathsig to kill subprocess if playground dead

### LINKED ISSUE #720: set Pdeathsig as KILL for all Command start by playground
## Feature Request

**Is your feature request related to a problem? Please describe:**

- start playground
- kill -9 playground
All the subprocess of playground orphaned child process adopted by init.

**Describe the feature you'd like:**

We can set Pdeathsig(Linux only) as KILL for all Command start by the playground to kill all processes.
ref: https://golang.org/pkg/syscall/#SysProcAttr
and https://medium.com/@ganeshmaharaj/clean-exit-of-golangs-exec-command-897832ac3fa5

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by july2993**
### What problem does this PR solve? 
fix #720 
Set Pdeathsig to kill subprocess if playground dead
### What is changed and how it works?
After this pr, if playground dead and some subprocess still running,
the subprocess will be killed by OS instead of becoming an orphaned child
adopted by init. This feature is linux only.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
before this pr:
Screen Shot 2020-08-31 at 2 57 45 PM
after this pr:

Release notes:

```release-note
Set Pdeathsig to kill subprocess if playground dead
```

### CODE REVIEW

File: pkg/exec/run.go [hunk 1]
```diff
@@ -137,6 +137,7 @@ func PrepareCommand(
 	version v0manifest.Version,
 	binPath, tag, instanceDir string, wd string,
 	args []string,
+	sysProcAttr *syscall.SysProcAttr,
```
Comment (lonng) [inline pkg/exec/run.go:140]: The function `PrepareCommand` has 11 parameters already, it's better to use a struct to define. How do you think?

**Review by lonng — APPROVED** [review]
Rest LGTM

**Comment by lonng** [comment]
Is any solution for macOS to handle the same scenario?

**Comment by july2993** [comment]
> Is any solution for macOS to handle the same scenario?

No as far as I know.

File: pkg/exec/run.go (hunk 1 above)
Comment (july2993) [reply pkg/exec/run.go]: https://github.com/pingcap/tiup/pull/733/commits/3ddcf04d4ad043213bc46ad80855398018a92701

**Comment by lucklove** [comment]
LGTM

Maybe it's better to add related issue into a milestone before we starting work on that... Otherwise the workload may be out of control.

**Pull request merged**